src/ray/core_worker/core_worker.cc

```diff
@@ -728,32 +749,31 @@ bool CoreWorker::AddActorHandle(std::unique_ptr<ActorHandle> actor_handle) {
     // Register a callback to handle actor notifications.
     auto actor_notification_callback = [this](const ActorID &actor_id,
                                               const gcs::ActorTableData &actor_data) {
-      if (actor_data.state() == gcs::ActorTableData::RECONSTRUCTING) {
-        auto it = actor_handles_.find(actor_id);
-        RAY_CHECK(it != actor_handles_.end());
-        if (it->second->IsDirectCallActor()) {
-          // We have to reset the actor handle since the next instance of the
-          // actor will not have the last sequence number that we sent.
-          // TODO: Remove the check for direct calls. We do not reset for the
-          // raylet codepath because it tries to replay all tasks since the
-          // last actor checkpoint.
-          it->second->Reset();
-        }
-      } else if (actor_data.state() == gcs::ActorTableData::DEAD) {
+      RAY_CHECK(actor_data.state() != gcs::ActorTableData::RECONSTRUCTING);
```

Comment:
if we are no longer publishing `RECONSTRUCTING` state, then how about the tasks with pending replies? Sounds like we need to re-send them (or otherwise fail them, but in that case we will need this reonstructing state).

Reply:
@ericl  Would the pending tasks be handled correctly if we don't process RECONSTRUCTING state?

Reply:
I don't think direct call actors can ever get in this state for now.